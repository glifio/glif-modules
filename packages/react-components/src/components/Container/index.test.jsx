import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import { Base } from './index.stories'

describe('Container', () => {
  afterEach(cleanup)
  test('renders the container', () => {
    const { container } = render(<Base {...Base.args} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders the container with the right attributes', () => {
    const { container } = render(<Base {...Base.args} />)

    const box = container.querySelector('div')
    expect(box).toHaveStyle('width: auto;')
    expect(box).toHaveStyle('position: flex;')
    expect(box).toHaveStyle('border: 1;')
    expect(box).toHaveStyle('align-items: center;')
    expect(box).toHaveStyle('justify-content: space-between;')
    expect(box).toHaveStyle('background-color: blue;')
    expect(box).toHaveStyle('padding: 4px;')
  })
})
