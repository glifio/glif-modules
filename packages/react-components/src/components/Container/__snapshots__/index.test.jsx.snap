// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders the container 1`] = `
.c0 {
  min-width: 0;
  box-sizing: border-box;
  position: flex;
  border: 1;
  padding: 4px;
  background-color: blue;
  width: auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@media screen and (min-width:40em) {
  .c0 {
    padding: 16px;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    padding: 32px;
  }
}

<div
  class="c0"
  width="auto"
>
  <div>
    Hello world
  </div>
</div>
`;
